src/network/networkd-ndisc.c

```diff
@@ -839,6 +839,13 @@ static int ndisc_router_process_route(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Failed to get gateway address from RA: %m");
 
+        if (link_has_ipv6_address(link, &gateway) == 0) {
+                _cleanup_free_ char *buf = NULL;
+                (void) in_addr_to_string(AF_INET6, (const union in_addr_union*) &gateway, &buf);
```

Comment:
Please use `if (DEBUG_LOGGING)`.

Please drop unnecessary type cast. Instead of casting, please change the type of gateway to in_addr_union.

Also, please add one empty line after the variable declaration.

Reply:
Thank you. All done.